src/v3/src/components/Radio/Radio.tsx

```diff
@@ -54,91 +43,42 @@ const Radio: UISchemaElementComponent<UISchemaElementComponentWithValidationProp
   const focusRef = useAutoFocus<HTMLInputElement>(focus);
-  const hasErrors = typeof errors !== 'undefined';
-  const tokens = useOdysseyDesignTokens();
+  const { errorMessage, errorMessageList } = buildFieldLevelErrorMessages(errors);
 
   return (
-    <FormControl
-      component="fieldset"
-      error={hasErrors}
+    <RadioGroup
+      errorMessage={errorMessage}
+      errorMessageList={errorMessageList}
+      hint={!required ? optionalLabel : undefined}
+      id={name}
+      label={label ?? ''}
+      name={name}
+      onChange={(e: ChangeEvent<HTMLInputElement>) => {
+        handleChange?.(e.currentTarget.value);
+      }}
+      testId={name}
+      value={value as string ?? ''}
     >
-      {label && (
-        <FormLabel
-          // To prevent asterisk from shifting far right
-          sx={{ display: 'flex', justifyContent: showAsterisk ? 'flex-start' : 'space-between' }}
-        >
-          {label}
-          {showAsterisk && (
-            <Box
-              component="span"
-              sx={{
-                marginInlineStart: tokens.Spacing2,
-                marginInlineEnd: tokens.Spacing2,
-              }}
-              className="no-translate"
-              aria-hidden
-            >
-              *
-            </Box>
-          )}
-          {required === false && (
-            <Typography
-              variant="subtitle1"
-              sx={{ whiteSpace: 'nowrap' }}
-            >
-              {optionalLabel}
-            </Typography>
-          )}
-        </FormLabel>
-      )}
-      <RadioGroup
-        name={name}
-        id={name}
-        data-se={name}
-        aria-describedby={describedByIds}
-        value={value as string ?? ''}
-        onChange={(e: ChangeEvent<HTMLInputElement>) => {
-          handleChange?.(e.currentTarget.value);
-        }}
-      >
-        {
-          (customOptions ?? options)?.map((item: IdxOption, index: number) => (
-            <FormControlLabel
-              control={(
-                <RadioMui
-                  sx={{
-                    marginInlineEnd: tokens.Spacing3,
-                  }}
-                />
-              )}
-              key={item.value}
-              value={item.value}
-              label={item.label}
-              disabled={loading}
-              onBlur={(e: ChangeEvent<HTMLInputElement>) => {
-                handleBlur?.(e?.currentTarget?.value);
-              }}
-              sx={{
-                gap: 0,
-              }}
-              // eslint-disable-next-line react/jsx-props-no-spreading
-              {...(index === 0 && { inputRef: focusRef })}
-            />
-          ))
-        }
-      </RadioGroup>
-      {hasErrors && (
-        <FieldLevelMessageContainer
-          messages={errors}
-          fieldName={name}
-        />
-      )}
-    </FormControl>
+      {
+        (customOptions ?? options)?.map((item: IdxOption, index: number) => (
```

Comment:
```suggestion
        (customOptions ?? options ?? []).map((item: IdxOption, index: number) => (
```
This will make linter happy , without typecasting to `JSX.Element[]`